scripts/sentry-xcodebuild.sh

```diff
@@ -57,63 +63,122 @@ fi
+    RUN_TEST_WITH_TSAN=false
+    ;;
+"tsan")
+    RUN_ANALYZE=false
+    RUN_BUILD=false
+    RUN_BUILD_FOR_TESTING=false
+    RUN_TEST_WITHOUT_BUILDING=false
+    RUN_TEST_WITH_TSAN=true
     ;;
 *)
+    RUN_ANALYZE=false
     RUN_BUILD=false
     RUN_BUILD_FOR_TESTING=true
     RUN_TEST_WITHOUT_BUILDING=true
+    RUN_TEST_WITH_TSAN=false
     ;;
 esac
 
-if [ "$CI" == true ]; then
+if [ "${CI-false}" == true ]; then
     RUBY_ENV_ARGS=""
 else
     RUBY_ENV_ARGS="rbenv exec bundle exec"
 fi
 
 if [ $RUN_BUILD == true ]; then
-    set -o pipefail && env NSUnbufferedIO=YES xcodebuild \
+    set -o pipefail && env NSUnbufferedIO=YES CODE_SIGN_IDENTITY=\"\" CODE_SIGNING_REQUIRED=NO xcodebuild \
         -workspace Sentry.xcworkspace \
-        -scheme "$TEST_SCHEME" \
+        -scheme "$SENTRY_SCHEME" \
         -configuration "$CONFIGURATION" \
         -destination "$DESTINATION" \
         -derivedDataPath "$DERIVED_DATA_PATH" \
         -quiet \
         build \
-        | xcpretty
-fi
-
-if [ $RUN_BUILD_FOR_TESTING == true ]; then
-    set -o pipefail && env NSUnbufferedIO=YES xcodebuild \
+        | $RUBY_ENV_ARGS xcpretty
+elif [ $RUN_ANALYZE == true ]; then
+    rm -rf analyzer
+    set -o pipefail && env NSUnbufferedIO=YES CODE_SIGN_IDENTITY=\"\" CODE_SIGNING_REQUIRED=NO xcodebuild \
         -workspace Sentry.xcworkspace \
-        -scheme "$TEST_SCHEME" \
+        CLANG_ANALYZER_OUTPUT=html \
+        CLANG_ANALYZER_OUTPUT_DIR=analyzer \
+        CODE_SIGNING_ALLOWED="NO" \
+        -scheme "$SENTRY_SCHEME" \
         -configuration "$CONFIGURATION" \
-        -destination "$DESTINATION" -quiet \
-        build-for-testing \
-        | xcpretty
-fi
-
-if [ $RUN_TEST_WITHOUT_BUILDING == true ]; then
-    set -o pipefail && env NSUnbufferedIO=YES && set -o pipefail && xcodebuild \
+        -destination "$DESTINATION" \
+        analyze \
+        2>&1 \
+        | $RUBY_ENV_ARGS xcpretty -t \
+        && [[ -z $(find analyzer -name "*.html") ]]
+elif [ $RUN_TEST_WITH_TSAN == true ]; then
+    # When enableThreadSanitizer is enabled and ThreadSanitizer finds an issue,
+    # the logs only show failing tests, but don't highlight the threading issues.
+    # Therefore we print a hint to find the threading issues. Profiler doesn't
+    # run when it detects TSAN is present, so we skip those tests.
+    set -o pipefail && env NSUnbufferedIO=YES CODE_SIGN_IDENTITY=\"\" CODE_SIGNING_REQUIRED=NO xcodebuild \
         -workspace Sentry.xcworkspace \
-        -scheme "$TEST_SCHEME" \
+        -scheme "$SENTRY_SCHEME" \
         -configuration "$CONFIGURATION" \
+        -enableThreadSanitizer YES \
         -destination "$DESTINATION" \
-        test-without-building \
-        | tee raw-test-output.log \
-        | $RUBY_ENV_ARGS xcpretty -t \
-        && slather coverage --configuration "$CONFIGURATION"
+        -skip-testing:"SentryProfilerTests" \
+        test \
+        | tee thread-sanitizer.log \
+        | xcpretty -t
+
+    testStatus=$?
```

Comment:
`m`: Correct me if I am wrong, but I think this will not be executed because the entire script will exit if the xcodebuild command fails

Reply:
IDK, i moved this from https://github.com/getsentry/sentry-cocoa/pull/4818/files#diff-b3aa5b62d9aab3e854e3fe47cc6aaacd2e4bd86fd6bb7484fb51d698623bf6a6L8-L19

Reply:
Could it be that it used to work, but then the addition of `&& set -o pipefail` at the beginning of the pipeline would cause it to no longer propagate the `$?` the same way? we also already had that at the top of the script, but i think that wouldn't have applied to the pipeline, which is why it was added there?